src/IonReader.ts

```diff
@@ -152,4 +152,14 @@ export interface Reader {
      * @throw Error when the reader is not positioned on a scalar value.
      */
     value(): ReaderScalarValue;
-}
\ No newline at end of file
+
+    /**
+     * Returns a representation of the ion document.
+     * 
+     * @return `string` String value when text.
+     * @returns `Uint8Array` Binary array when binary.
+     * 
+     * @throws
+     */
+    deepValue: () => string | Uint8Array;
```

Comment:
I am a little confused as to why this is called `deepValue()` since in theory it could be called at an point in the `Reader` including the top-level.  Also, the doc comment really should indicate the caveats to this API.  For example, I *assume* (though not sure it is the intent), that the local symbol table context is not necessarily available using this API and the caller needs to be aware of such things.  Is that correct?

Reply:
I think you are correct. I meant "deep" because it goes into the structure recursively.